content_handler/session_connection.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/session_connection.h"
+#include "apps/mozart/lib/scene/session_helpers.h"
+
+namespace flutter_runner {
+
+SessionConnection::SessionConnection(mozart2::SessionPtr session,
+                                     mx::eventpair import_token)
+    : session_(std::move(session)), root_node_(&session_) {
+  root_node_.Bind(std::move(import_token));
+  session_.set_connection_error_handler(
+      std::bind(&SessionConnection::OnSessionError, this));
+  present_callback_ =
+      std::bind(&SessionConnection::OnPresent, this, std::placeholders::_1);
+}
+
+SessionConnection::~SessionConnection() {
+  session_.set_connection_error_handler(nullptr);
+}
+
+mozart::client::ImportNode& SessionConnection::GetRootNode() {
+  return root_node_;
+}
+
+mozart::client::Session& SessionConnection::GetSession() {
+  return session_;
+}
+
+void SessionConnection::OnSessionError() {
+  // TODO: Not this.
+  FTL_CHECK(false) << "Session connection was terminated.";
+}
+
+void SessionConnection::PresentFrameOpsNow() {
+  session_.Present(0,                 // presentation_time. Placeholder for now.
+                   present_callback_  // callback
+                   );
+  EnqueueClearOps();
+}
+
+void SessionConnection::OnPresent(mozart2::PresentationInfoPtr info) {
+  FTL_LOG(INFO) << "Session presentation did complete.";
+}
+
+void SessionConnection::EnqueueClearOps() {
+  // We are going to be sending down a fresh node hierarchy every frame. So just
+  // enqueue a detach op on the imported root node.
+  session_.Enqueue(mozart::NewDetachChildrenOp(root_node_.id()));
```

Comment:
unfortunately this op isn't implemented, I recommend creating a throwaway node to act as an intermediate, to clear the frame, detach that node and make a new one